### PR TITLE
Plat admin setup

### PR DESCRIPTION
**Pull request by izaxveta**
#### Pivotal URL: 
https://www.pivotaltracker.com/story/show/153655360

#### What does this PR do?
Enables Platform Admin to log in and be directed to appropriate dashboard

#### Where should the reviewer start?
spec/features/admin/platform_admin_can_sign_in_spec.rb

#### How should this be manually tested?
All tests passing
image

#### Any background context you want to provide?
A platform admin should be able to log in and see indication of their role and be routed to admin dashboard. No view dependencies or limitations based on user role have been implemented yet.

#### Screenshots (if appropriate)
image
🎉🎉

#### Questions:
  - Do Migrations Need to be ran?
Very much so.
  - Do Environment Variables need to be set?
No
  - Any other deploy steps?
You may or may not need to run `Role.create(name: "Platform Admin")` within `$ rails c`

### CODE REVIEW

File: config/routes.rb [hunk 1]
```diff
@@ -18,7 +18,7 @@
   end
 
 
-  resources :users , only: [:new, :create, :edit, :update]
+  resources :users , only: [:new, :create, :update]
```
Comment (marshallhouston) [inline config/routes.rb]: This change doesn't seem related to a platform_admin logging in.

File: spec/features/user/user_edits_credentials_spec.rb [hunk 2]
```diff
@@ -0,0 +1,28 @@
+require 'rails_helper'
```
Comment (marshallhouston) [inline spec/features/user/user_edits_credentials_spec.rb]: I don't see anything about a platform admin here. I think this is related to a different story and shouldn't be in this PR.

File: config/routes.rb [hunk 3]
```diff
@@ -37,4 +37,6 @@
 
   resources :categories, only: [:show], param: :category
 
+  get '/account/edit', to: 'users#edit', as: 'edit_user'
```
Comment (marshallhouston) [inline config/routes.rb]: Is this related to the platform admin?

File: app/controllers/admin/dashboard_controller.rb [hunk 4]
```diff
@@ -7,6 +7,6 @@ def index
     else
       @orders = Order.all
     end
-    flash[:notice] = "You're logged in as an Administrator."
+    flash[:notice] = "You're logged in as #{current_user.highest_role}."
```
Comment (marshallhouston) [inline app/controllers/admin/dashboard_controller.rb:10]: I like that this is dynamic.

File: app/models/user.rb [hunk 5]
```diff
@@ -11,6 +13,11 @@ def full_name
     first_name + " " + last_name
   end
 
+  def highest_role
+    return "a #{roles.last.name}" if !roles.empty?
```
Comment (marshallhouston) [inline app/models/user.rb:17]: A possible refactor.

if roles.empty?
 "an Administrator"
else
 "a #{roles.last.name}"
end

File: app/controllers/users_controller.rb [hunk 6]
```diff
@@ -21,7 +21,8 @@ def update
       redirect_to admin_dashboard_index_path
     elsif current_user != nil
       current_user.update(user_params)
-      redirect_to dashboard_index_path
+      flash[:success] = "Successfully updated your account information."
```
Comment (marshallhouston) [inline app/controllers/users_controller.rb]: seems unrelated to the platform_admin

**Review by marshallhouston — CHANGES_REQUESTED** [review]
The work on the Platform Admin section looks great, but it looks like there are changes that are not related to platform admin and are related to the story about a user editing account info.

**Review by adrian-lara — DISMISSED** [review]
I agree with Marshall in that the "edit user credentials" work should not be included in this PR.

File: app/models/user.rb (hunk 5 above)
Comment (izaxveta) [reply app/models/user.rb:17]: This will change further down the line when we begin to incorporate other role types. For right now there's only 'Platform Admin'. If a user.roles is empty, then that's indicative that they're not an admin of any sort. If user.roles is _not_  empty, then they're an admin as the only role that exists is Platform Admin. This will become more complex and specific once we begin to add more roles.

**Review by marshallhouston — APPROVED** [review]
